src/decisionengine/framework/dataspace/datasources/tests/test_datasource_api.py

```diff
@@ -419,7 +419,8 @@ def test_update_bad(datasource):  # noqa: F811
         generation_id=1,
         key="my_test_key",
     )
-    with pytest.raises((KeyError, NoResultFound)):
+
+    with pytest.raises(Exception):
```

Comment:
Which other exception are you expecting? Should there be a pragma to avoid flagging?

Reply:
This will produce some sort of DB-API Constraint Failure.  The exact exception depends on the backend in use.  For `psycopg2` I'd expect `psycopg2.errors.IntegrityConstraintViolation` or perhaps `IntegrityError`



`